provider/postgres/postgres.go

```diff
@@ -0,0 +1,661 @@
+// Copyright 2015 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	_ "github.com/lib/pq"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+)
+
+// SQLNotifies implements a Notifies provider based on a SQL DB.
+type SQLNotifies struct {
+	db *sql.DB
+}
+
+// NewSQLNotifies returns a new SQLNotifies based on the provided SQL DB.
+func NewPostgresNotifies(db *sql.DB) (*SQLNotifies, error) {
+	tx, err := db.Begin()
+	if err != nil {
+		return nil, err
+	}
+	if _, err := tx.Exec(createNotifyInfoTable); err != nil {
+		tx.Rollback()
+		return nil, err
+	}
+
+	// XXX(bug-ish): The selection of pending alerts uses a NOT IN clause
```

Comment:
All this noise below was to work around QL behavior I couldn't really explain myself. Probably not necessary for Postgres though, right?